certbot-nginx/certbot_nginx/parser.py

```diff
@@ -536,8 +515,10 @@ def _add_directive(block, directive, replace):
 
     # Find the index of a config line where the name of the directive matches
     # the name of the directive we want to add. If no line exists, use None.
-    location = next((index for index, line in enumerate(block)
-                     if line and line[0] == directive[0]), None)
+    find_location = lambda direc: next((index for index, line in enumerate(block) \
```

Comment:
nit: perhaps you should make a small helper function for this as this is a bit complicated to read.